wave_kinematics/include/wave/kinematics/rotation.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef __WAVE_KINEMATICS_ROTATION_HPP__
+#define __WAVE_KINEMATICS_ROTATION_HPP__
+
+#include <iostream>
+#include <functional>
+
+#include "wave/utils/utils.hpp"
+#include <kindr/Core>
+
+namespace wave {
+
+class Rotation {
+public:
+  // Internal storage as a kindr Rotation object.
+  kindr::RotationMatrixD rotation_object;
```

Comment:
That's a tough question. Yes, if you want different function names and signatures you will have to wrap everything.

Right now, the difficulty of designing a good interface seems much greater than the difficulty of changing client calls. If there were 10 third party libraries depending on the `Rotation` class that would be different, but I don't see how we'd get to that point without a really solid, versatile interface. And the law of leaky abstractions suggests that we would have to change some client code, anyway.

For example, Kindr itself wraps Eigen objects but it takes thousands of lines of code to do so.